samza-core/src/main/java/org/apache/samza/coordinator/leaderelection/LeaderElector.java

```diff
@@ -28,7 +30,7 @@
    *
    * @return True, if caller is chosen as a leader through the leader election process. False, otherwise.
    */
-  boolean tryBecomeLeader();
+  boolean tryBecomeLeader(LeaderElectorListener leaderElectorListener);
```

Comment:
Ok. I still don't see how tryBecomeLeader is async. If it were async, the caller invoking tryBecomeLeader should return before the leader election is complete. In the current implementation, it is not async. 
So, my question is: What is need for this API to be async? Is this a requirement from Coordination Service ?